src/freenas/usr/local/bin/truenas-grub.py

```diff
@@ -19,18 +20,24 @@ def get_serial_ports():
     # 22.02-RC.2 use real hostid. We need to be able to boot both of these configurations.
     config = [
         'GRUB_DISTRIBUTOR="TrueNAS Scale"',
+        'GRUB_TIMEOUT=10',
         'GRUB_CMDLINE_LINUX_DEFAULT="libata.allow_tpm=1 amd_iommu=on iommu=pt '
         'kvm_amd.npt=1 kvm_amd.avic=1 intel_iommu=on zfsforce=1'
         f'{f" {kernel_extra_options}" if kernel_extra_options else ""}"',
     ]
 
-    terminal = ["console"]
+    terminal_output = ["console"]
+    terminal_input = ["console"]
     cmdline = []
     if advanced["serialconsole"]:
-        config.append(f'GRUB_SERIAL_COMMAND="serial --speed={advanced["serialspeed"]} --word=8 --parity=no --stop=1"')
-        terminal.append("serial")
         port = get_serial_ports().get(advanced['serialport'], advanced['serialport'])
-        cmdline.append(f"console={port},{advanced['serialspeed']} console=tty1")
+        port_nr = port.replace('ttyS', '')
+        config.append(f'GRUB_SERIAL_COMMAND="serial --unit={port_nr} --speed={advanced["serialspeed"]} --word=8 --parity=no --stop=1"')
+        terminal_output.append("serial")
+        terminal_input.append("serial")
+        cmdline.append(f"console=tty1 console={port},{advanced['serialspeed']}")
+        if os.path.exists("/sys/firmware/efi"):
+            terminal_output[0] = "gfxterm"
```

Comment:
It is probably fine, but it looks dirty to me to use both append() and direct indexes together.  I would probably move this "if" block before the terminal_output.append("serial") and do terminal_output = ["gfxterm"] instead.

Reply:
Thank you. I have updated it as you suggested.